fieldservice_size/models/fsm_size.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2020 Brian McMaster <brian@mcmpest.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class FSMSize(models.Model):
+    _name = 'fsm.size'
+    _description = 'Field Service Size'
+
+    name = fields.Char(string='Name', required='True')
+    type_id = fields.Many2one('fsm.order.type', string='Order Type')
+    parent_id = fields.Many2one('fsm.size', string='Parent Size', index=True)
+    uom_id = fields.Many2one('uom.uom', string='Unit of Measure')
+    is_order_size = fields.Boolean(
+        string='Is the Order Size?',
+        help='The default size for orders of this type')
+
+    _sql_constraints = [
+        ('name_uniq', 'unique (name)', "Size name already exists!"),
+    ]
+
+    @api.constrains('is_order_size', 'type_id')
+    def _one_size_per_type(self):
+        size = self.search([
```

Comment:
Not a requested change, but you may use search_count to get better perfs

Reply:
Great. Thank you!